superset/db_engine_specs/trino.py

```diff
@@ -227,12 +227,22 @@ def execute_with_cursor(
         execute_event = threading.Event()
 
         def _execute(
-            results: dict[str, Any], event: threading.Event, app: Flask
+            results: dict[str, Any],
+            event: threading.Event,
+            app: Flask,
+            g_copy: ctx._AppCtxGlobals,
         ) -> None:
             logger.debug("Query %d: Running query: %s", query_id, sql)
 
             try:
+                # Flask contexts are local to the thread that handles the request.
+                # When you spawn a new thread, it does not inherit the contexts
+                # from the parent thread,
+                # meaning the g object and other context-bound variables are not
+                # accessible
                 with app.app_context():
```

Comment:
@dpgaspar I did a quick GPT search.

```
from flask import current_app
import threading

def my_thread():
    with current_app.app_context():
        # Access Flask context within the thread
        app = current_app._get_current_object()
        # Do something with the app

# Create and start the thread
thread = threading.Thread(target=my_thread)
thread.start()
```

Would something like this work for our use cases?

Reply:
seems basically similar, except I'm passing the current_app into the thread, also we need to recreate g inside the thread

Reply:
Can't we use [copy_current_request_context](https://tedboy.github.io/flask/generated/flask.copy_current_request_context.html)?

```
@copy_current_request_context
def worker():
    # Access Flask's g object here
    # Your code goes here
```

Reply:
only the request context is copied according to: https://github.com/pallets/flask/issues/3306